src/lib/signature_lib/keypair.mli

```diff
@@ -5,14 +5,15 @@ module Stable : sig
     type t =
       { public_key: Public_key.Stable.V1.t
       ; private_key: Private_key.Stable.V1.t sexp_opaque }
-    [@@deriving sexp, bin_io]
+    [@@deriving sexp, bin_io, version]
   end
 
   module Latest = V1
 end
 
 type t = Stable.V1.t =
-  {public_key: Public_key.t; private_key: Private_key.t sexp_opaque}
+  { public_key: Public_key.Stable.V1.t
```

Comment:
I think this (and the below) should refer to the un-versioned `Public_key.t`, this doesn't seem like it has anything to do with versioning.

Reply:
Usually, I've been doing this:
```
  type t = Stable.Latest.t = ...
```
where `...` is exactly the way `Stable.Latest.t` is defined, which mentions specific versions.

We don't know that the unversioned types are always the same type as the versions mentioned. If `Public_key` evolves, then `Public_key.t` may differ from `Public_key.Stable.V1.t`, for example.

So I think `Stable.V1` in this code should be `Stable.Latest`, and the versions should be mentioned in the type definition.

But maybe I'm misunderstanding ...

Reply:
@mrmr1993 Thought about this some more, and indeed, the habit I'd fallen into is not the best. If `Stable.Latest` is using types that have since evolved, then a new version needs to be added to `Stable`.

The linter feature that detects uses of versioned types outside of versioned type definitions will catch these.